### PR TITLE
Fixes/UUID

### PR DESCRIPTION
**Pull request by toitzi**
# ORM Behavior Updates and Fixes

### Description

Fixing Issue #288 by adding UUID class (original from Kohana ported to Koseven)
Also "combining" PR #244 (Added option to switch off the verification of the GUID(/UUID), when inserting a lot of columns this check decreased the insert performance.) and PR #290 (fix UUID not exist) in this one

### Related Issue

#288 from @WinterSilence 

### How Has This Been Tested

Unit Tests have been added and code has been tested by @toitzi and shadowhand (back in Kohana times)

### Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by toitzi** [comment]
Can I merge this?

File: modules/orm/classes/ORM/Behavior/Guid.php [hunk 1]
```diff
@@ -15,105 +14,109 @@ class ORM_Behavior_Guid extends ORM_Behavior {
...
-        $model->where($this->_guid_column, '=', $id)->find();
-
-        // Prevent further record loading
-        return FALSE;
-      }
-    }
-
-    return TRUE;
+		if (($id !== NULL) && ! is_array($id) && ! ctype_digit($id)) {
+			if (UUID::valid($id)) {
```
Comment (errotan) [inline modules/orm/classes/ORM/Behavior/Guid.php]: This check can be put in the upper if.

Comment (toitzi) [reply modules/orm/classes/ORM/Behavior/Guid.php]: Resolved Thanks!

**Comment by toitzi** [comment]
> Guid verify and namespace check must be mandatory and throw an exception on failure. Returning FALSE may hide bad functionality.

Good point, updated code and tests to throw Exception if invalid UUID or invalid Namespace detected!

**Comment by piotrbaczek** [comment]
I mean this is fine, however there are already out there some well-maintained libraries that provide the same functionality, f.ex. https://github.com/ramsey/uuid that has 8000+ stars.
I mean what you did is fine, however for future reference I would like to not re-invent the wheel.

Good job though @toitzi

**Pull request merged**